compiler/qsc_parse/src/lex/concrete.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use std::iter::Peekable;
+
+use qsc_data_structures::span::Span;
+
+use super::{cooked, raw};
+
+pub(super) struct ConcreteToken {
+    pub kind: ConcreteTokenKind,
+    pub span: Span,
+}
+
+#[derive(Debug, PartialEq)]
+pub(super) enum ConcreteTokenKind {
+    Cooked(cooked::TokenKind),
```

Comment:
The way this enum is organized is throwing me off a bit at the usage site (in the bit match statement in formatter.rs). 

Instead of "leaking" the low-level concept of cooked tokens, I'd hope to see this enum organized a bit more abstractly, like:

```rs
pub(super) enum ConcreteTokenKind {
    Syntax(cooked::TokenKind),  // nontrivial tokens. Not sure if this is the best name to use. Maybe `Compilation`?
    Error(cooked::Error),
    Whitespace,
    Comment // this would INCLUDE doc comments, since you don't actually need the differentiate them from regular comments for formatting
}

Reply:
Discussed offline: The doc comments are logically different from regular comments and are compiled tokens, so it makes sense to keep them separate. Renaming the "Cooked" case makes sense and will be done.

Reply:
Renamed to `Syntax`